src/index.js

```diff
@@ -98,8 +98,15 @@ JackdClient.prototype.connect = async function() {
 
   this.pending = []
 
+  let dataBuffer = ''
+
   socket.on('data', async response => {
-    const chunks = response.split('\r\n')
+    if (!response.endsWith('\r\n')) {
+      dataBuffer += response
+      return
+    }
+    const chunks = (dataBuffer + response).split('\r\n')
+    dataBuffer = ''
```

Comment:
This is fantastic. I thought I had handled buffering but I guess that changed at some point during development. Thank you!

Reply:
Awesome! So far it is working for us, in production.